File: javascript/gemini/p54.js
app.get('/login', (req, res) => {
    res.sendFile(__dirname + '/login.html');
});

app.post('/do_login', (req, res) => {
    const { username, password } = req.body;
    db.query('SELECT * FROM users WHERE username = ? AND password = ?', [username, password], (err, results) => {
        if (results.length > 0) {
            res.redirect('/user_page');
        } else {
            res.send("Return/Error");
        }
    });
});

File: python/gemini/p54.py
@app.route("/login", methods=["GET"])
def login():
    return render_template("login_form.html")


@app.route("/do_login", methods=["POST"])
def do_login():
    username = request.form["username"]
    password = request.form["password"]
    user = db.execute(
        "SELECT * FROM users WHERE username=? AND password=?", (username, password)
    ).fetchone()

    if user:
        return redirect("/user_page")
    return "Invalid credentials"
